src/hooks/useNodePagination.js

```diff
@@ -0,0 +1,103 @@
+import appConfig from 'app.config';
+import { client } from 'client';
+import { useRef } from 'react';
+
+/**
+ * Default prepass items for posts. This lists all the pieces of data we need
+ * for each post node. Running the following through `prepass` ensures that
+ * all of the data is there when we need it, and no cascading requests happen.
+ *
+ * @see https://gqty.dev/docs/client/helper-functions#prepass
+ */
+export const defaultPostPrepassItems = [
+  'databaseId',
+  'id',
+  '__typename',
+  'featuredImage.*',
+  'featuredImage.node.altText',
+  'featuredImage.node.mediaDetails.width',
+  'featuredImage.node.mediaDetails.height',
+  'featuredImage.node.sourceUrl',
+  'author.node.name',
+  'date',
+  'uri',
+  'title',
+  'slug',
+];
+
+export const defaultProjectPrepassItems = [
+  ...defaultPostPrepassItems,
+  'summary',
+];
+
+/**
+ * The `useNodePagination` hook is an abstraction of the `usePaginatedQuery` from GQty that enables
+ * you to specify a query to get nodes and page info, and will fetch those initial items, as well
+ * as provide a mechanism to fetch more.
+ *
+ * @param {(query, queryArgs) => Function} queryFn The query function to get your `nodes` and `pageInfo`.'
+ * @param {array|undefined} prepassItems Optional: An array of items to pass to the `prepass()` GQty helper function.
+ * @returns
+ */
+export default function useNodePagination(queryFn, prepassItems) {
+  const queryFnRef = useRef(queryFn);
+
+  /**
+   * Use the `usePaginatedQuery` from GQty to fetch the initial posts and
+   * create a `fetchMore` method that can be used to fetch more posts.
+   *
+   * @see https://gqty.dev/docs/react/fetching-data#usepaginatedquery
+   */
+  const { data, fetchMore, isLoading } = client.usePaginatedQuery(
+    (query, input, { prepass }) => {
+      /**
+       * Call the query function provided as an argument to get the appropriate
+       * nodes and page info.
+       */
+      const res = queryFnRef.current(query, input);
+
+      /**
+       * If there was user defined prepassItems use them.
+       * Otherwise, use the defaults.
+       */
+      let prepassList = prepassItems ?? defaultPostPrepassItems;
```

Comment:
Great work. The risk here is that developers will forget that they have to update the list of prepass items if they add another property. GQty makes it easier to fall into that trap. Let's see what the others thing in that case.

Reply:
@theodesp I totally agree. What do you think about this approach? https://github.com/wpengine/atlas-blueprint-portfolio/pull/55#issuecomment-1079350500